keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -931,28 +716,27 @@ Several components must be implemented or modified in Kubernetes:
   ResourceClaim (directly or through a template) and ensure that the
   resource is allocated before the Pod gets scheduled, similar to
   https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/scheduling/scheduler_binder.go
-- Kubelet must be extended to retrieve information from ResourceClaims
-  and to call a resource kubelet plugin. That plugin returns CDI device ID(s)
+- Kubelet must be extended to manage ResourceClaims
+  and to call a kubelet plugin. That plugin returns CDI device ID(s)
   which then must be passed to the container runtime.
 
-A resource driver can have the following components:
-- *CRD controller* (optional): a central component which translates parameters
-  defined with a vendor CRD into in-tree parameter types. 
-- *kubelet plugin* (required): a component which cooperates with kubelet to
-  publish resource information and to prepare the usage of the resource on a node.
+A DRA driver can have the following components:
+- *admission webhook* (optional): a central component which checks the opaque
+  configuration parameters in ResourceClaims, ResourceClaimTemplates and DeviceClasses at
+  the time that they are created. Without this, invalid parameters can only
+  be detected when a Pod is about to run on a node.
+- *kubelet plugin* (required): a component which publishes device information
+  and cooperates with kubelet to prepare the usage of the devices on a node.
 
-When a resource driver doesn't use its own CRD for parameters, the CRD controller
-is not needed and a ResourceClaim references ResourceClaimParameters directly.
-
-A [utility library](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/dynamic-resource-allocation) for resource drivers was developed.
+A [utility library](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/dynamic-resource-allocation) for DRA drivers was developed.
 It does not have to be used by drivers, therefore it is not described further
 in this KEP.
 
 ### State and communication
 
-A ResourceClaim object defines what kind of resource is needed and what
-the parameters for it are. It is owned by users and namespaced. Additional
-parameters are provided by a cluster admin in ResourceClass objects.
+A ResourceClaim object defines what devices are needed and what
+the parameters for them are. It is owned by users and namespaced. Additional
+parameters are provided by a cluster admin in DeviceClass objects.
 
 The ResourceClaim spec is immutable. The ResourceClaim
```

Comment:
Is this still true? I seem to remember there was some talk about making it mutable...

Reply:
I suggested it, but @thockin and @johnbelamaric didn't like that, so it's still immutable.